neal/sampler.py

```diff
@@ -97,75 +99,95 @@ def __init__(self):
-                by `beta_schedule_type`. Default range is set based on the total
-                bias associated with each node.
+                by ``beta_schedule_type``. Default range is set based on the 
+                total bias associated with each node.
 
             num_reads (int, optional, default=len(initial_states) or 1):
-                Number of reads. Each read is generated by one run of the simulated
-                annealing algorithm. If `num_reads` is not explicitly given, it is
-                selected to match the number of initial states given. If initial states
-                are not provided, only one read is performed.
-
-            num_sweeps (int, optional, default=1000):
-                Number of sweeps or steps.
-
-            beta_schedule_type (string, optional, default='geometric'):
-                Beta schedule type, or how the beta values are interpolated between
-                the given 'beta_range'. Supported values are:
-
-                * linear
-                * geometric
-
-            seed (int, optional):
-                Seed to use for the PRNG. Specifying a particular seed with a constant
-                set of parameters produces identical results. If not provided, a random seed
-                is chosen.
+                Number of reads. Each read is generated by one run of the 
+                simulated annealing algorithm. If `num_reads` is not explicitly
+                given, it is selected to match the number of initial states 
+                given. If initial states are not provided, only one read is 
+                performed.
+
+            num_sweeps (int, optional, default=``len(beta_schedule)*num_sweeps_per_beta`` or 1000):
+                Number of sweeps used in annealing. If no value is provided
+                and ``beta_schedule`` is None the value is defaulted to 1000.
+
+            num_sweeps_per_beta (int, optional, default=1)
+                Number of sweeps to perform at each beta. One sweep consists of
+                a sequential Metropolis update of all spins.
+
+            beta_schedule_type (string, optional, default="geometric"):
+                Beta schedule type, or how the beta values are interpolated 
+                between the given `beta_range`. Supported values are:
+                * "linear"
+                * "geometric"
+                * "custom"
+
+                "custom" is recommended for high-performance applications, which
+                typically require optimizing beta schedules beyond those of the
+                "linear" and "geometric" options, with bounds beyond those 
+                provided by default. ``num_sweeps_per_beta`` and 
+                ``beta_schedule`` fully specify a custom schedule.
+
+            beta_schedule (array-like, optional, default = None)
+                Sequence of beta values swept. Format compatible with 
+                numpy.array(beta_schedule,dtype=float) required. 
```

Comment:
Can we add the supported range here? I think that at a minimum values must be bigger than 0. Should the values be monotonically ascending? Etc.

Reply:
As per previous statement, there are exotic schedules like reverse annealing and bang-bang. Values of 0 and values of Inf work fine with the C code, and can be useful in special case. Negative values might cause numerical instability though, I didn't check this.

Reply:
I'll specify a positive range >= to be safe. There is not really a good use case for negative values